crates/circuit/src/circuit_data.rs

```diff
@@ -38,6 +44,13 @@ use smallvec::SmallVec;
 
 import_exception!(qiskit.circuit.exceptions, CircuitError);
 
+type CircuitDataState<'py> = (
```

Comment:
I'd throw a docstring on this to make it clear what it's for since it's about 150 lines away from the use:

```suggestion
/// A tuple of a `CircuitData`'s internal state used for pickle's `__setstate__()` method
type CircuitDataState<'py> = (
```

Reply:
Done in 0f8cf6593.